Converted market times sensor to ISO Date Time format & added AUD Currency

cabberley opened the pull request:
This update provides 2 enhancements to your code
1. Added AUD currency code so currency attribute will now correctly identify and display Australian stocks currencies. Previously it was displaying unknown.
2. The regularMarketTime & postMarketTime attributes were being displayed in Epoch, added code to convert these to a full DateTime with TimeZone aware value.

iprak commented on custom_components/yahoofinance/sensor.py:
```diff
@@ -25,9 +25,11 @@
     ATTR_CURRENCY_SYMBOL,
     ATTR_DIVIDEND_DATE,
-    ATTR_MARKET_STATE,
+    ATTR_MARKET_STATE,\
```
Can we remove the trailing \ ? Just to be consistent.

iprak commented on custom_components/yahoofinance/sensor.py:
```diff
@@ -178,6 +182,18 @@ def parse_dividend_date(dividend_date_timestamp) -> str | None:
+            return None
+
+        market_date = datetime.fromtimestamp(date_timestamp,tz=dt_util.DEFAULT_TIME_ZONE)
+        market_date_date = market_date  #.datetime()
```
We can just use `market_date` here.

iprak replied:
Also the new function is very much like `parse_dividend_date`. Can we refactor/rename that to handle instant case?

iprak commented on custom_components/yahoofinance/sensor.py:
```diff
@@ -178,6 +182,18 @@ def parse_dividend_date(dividend_date_timestamp) -> str | None:
         return dividend_date_date.isoformat()
+    
+    @staticmethod
+    def parse_market_time(date_timestamp) -> str | None:
```
Can we rename this to be more generic like convert_time_stamp_to_local ?

iprak reviewed (commented):
Some minor code comments. Thanks for the contribution.

cabberley commented:
@iprak Thanks for the feedback.
I have refactored the timestamp conversion and consolidated the two into a single one. The refactored also facilitates that Dividend Date is just a date no time. Market times, the date and time are relevant.

Also added the tests as requested.

iprak commented:
The combined function now returns 0 for missing (None) input. We should keep previous behavior of None. In cases where data might be missing, the result None can be treated differently than 0.

iprak commented:
Sorry but now a few different tests are failing. The tests are in test_sensor.py (test_parse_dividend_date) which is failing because it is using the older function.

cabberley commented:
reverted return value to be date_timestamp value to support the None and 0 use case. If 0 gets through and converted it would result in a 1970 date.
Rewrote the tests for the renamed function and added the extra testing to it to reflect the changes in the function.
**Sorry been a little while since I have done all of this. Hopefully this fixes it all up.

iprak commented on custom_components/yahoofinance/sensor.py:
```diff
@@ -179,7 +179,7 @@ def convert_timestamp_to_datetime(date_timestamp, return_format) -> str | None:
         date_timestamp = convert_to_float(date_timestamp)
         if date_timestamp is None or date_timestamp == 0:
-            return 0
+            return date_timestamp
```
For most symbols, I am seeing `0` as the value for preMarketTime in data which appears as 0. 

![image](https://github.com/iprak/yahoofinance/assets/6459774/76c63ddb-affc-4fbc-9554-fd1d956b9311)

If we adjusted convert_timestamp_to_datetime to return None , then it would appear as Unknown as for 
dividendDate. Yahoo is not providing any data for dividendDate in this case.


Thoughts? If we changed convert_timestamp_to_datetime, then the unit tests would also need some adjustment.

cabberley replied:
I agree that's why I changed the function to return the date_timestamp value. The float function returns either the numeric value which is either 0, epoch or None.if it is 0 or None then the if statement well return 0 or None. 

Shouldn't need any further changes.

iprak reviewed (commented):
Everything else looks good. One final comment.
